src/ChainSafe.Gaming/Web3/Core/LifecycleHandler.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace ChainSafe.Gaming.Web3.Core
+{
+    /// <summary>
+    /// Handles the lifecycle of <see cref="ILifecycleParticipant"/> instances.
+    /// </summary>
+    public class LifecycleHandler
```

Comment:
Looks more like `LifecycleManager` to me as it manages all the `ILifecycleParticipant`s.

Reply:
handler/manager seem interchangeable to me but if we have it like this elsewhere and we're sticking with uniformity then yeah, change it.

Reply:
handler, manager...potato, powtatow xD will change

Reply:
done